substrate/frame/broker/src/adapt_price.rs

```diff
@@ -17,58 +17,97 @@
 
 #![deny(missing_docs)]
 
-use crate::CoreIndex;
+use crate::SaleInfoRecord;
 use sp_arithmetic::{traits::One, FixedU64};
-use sp_runtime::Saturating;
+use sp_runtime::{FixedPointNumber, FixedPointOperand, Saturating};
+
+/// Performance of a past sale.
+#[derive(Copy, Clone)]
+pub struct SalePerformance<Balance> {
```

Comment:
I think that the new model is a step in the right direction, but now we should increase the number of data points. Instead of tracking only the most recent sellout value potentially track X recent sales and do a weighed average of them with recency being the weight. 

Such approaches can remove the unnecessary price and target restrictions while allowing for multiple data points which cannot be easily manipulated with individual purchases. Thoughts?

Reply:
Yea that's almost exactly what my hackmd about the pricing iteration talks about.
With a weighted average you can weight towards newer prices for the regions on the open market (to get better sensitivity to market forces), and weight towards the older data for renewals

Reply:
I mean just about the weighted average, but still using the full information of a sale (cores offered and sold contributes to the weighting)

Reply:
If I think more of it, I am not sure this is a real problem: It seems quite unrealistic that 49 cores are sold at a really high price and the last core at the minimum, but even in that extreme scenario the next sale will start at 2x that price.

Given that absolutely nobody wanted that last core, that 2x should be sufficient to reflect market well enough.

But in general handling a spike in supply (up and down) will obviously affect price. With the currently implemented model we can handle those very well if the go downwards and okayish when they go up - we adjust only 2x. But the only risk here would be to sell cores "too cheaply" for a short while. Assuming the price spike was just because of a short shortage on supply, I would argue that the moderate price increase is actually a good thing. 

We could make it symmetric: 10x up and 10x down from the target price - I actually already have that prepared, with the curve being adjusted so the optimal price is still in the middle time wise.

And in fact this might be better, it would still be expensive to do price manipulation and the system works against you. E.g. if at one sale only 1 core was offered and you bought it at the maximum price, the base price in the next sale would be that price/10 - which is exactly the target price of the previous sale. So actually no harm done.

Yeah. I think that change is actually good.